pkg/geo/geogfn/topology_operations.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geogfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/golang/geo/r3"
+	"github.com/golang/geo/s2"
+	"github.com/twpayne/go-geom"
+)
+
+// Centroid returns the Centroid of a given Geography.
+//
+// Note: In the case of (Multi)Polygon Centroid result, it doesn't mirror with
+// `PostGIS`'s result. Though we are using the same algorithm that is divide
+// the given region into the set of triangles and calculate the weighted average
+// of centroids in these regions as the resultant centroid. But `PostGIS`
+// implementation differs, that is, the method used for dividing the given region
+// into the set of triangles is different. In `PostGIS` implementation, one point
+// of the triangle (reference point) is fixed and other two points of the triangle
+// are changed by moving across various edges of various Ring in various Polygons.
+// Whereas in Centroid implementation, we always update the reference point when
+// moving from one Ring to another.
+//
+// Ideally, both the implementation should provide the same result. But since we are
+// using PlannerCentroid instead of actual projection in the Spherical surface,
```

Comment:
PlannerCentroid?

Reply:
Vectorized summation of all the points in `s2` is referred to as PlannerCentroid.
Updated it with: 
`the vectorized mean of all the points`